File: 0118-pascals-triangle/solution.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        pscl_trngl = []
        for i in range(numRows):
            row = [1] * (i + 1)
            for j in range(1, i):
                row[j] = pscl_trngl[i-1][j-1] + pscl_trngl[i-1][j]
            pscl_trngl.append(row)
        return pscl_trngl

File: 0118-pascals-triangle/solution.cpp
class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        vector<vector<int>> pscl_trngl;
        for (int i{0}; i < numRows; i++)
        {
            vector<int> row(i + 1, 1);
            for (int j{1}; j < i; j++)
            {
                row[j] = pscl_trngl[i - 1][j - 1] + pscl_trngl[i - 1][j];
            }
            pscl_trngl.emplace_back(row);
        }
        return pscl_trngl;
    }
};
